Project capstone

mandelatuks opened the pull request:
In this project update, I applied the skills learned throughout Module 1 of the Microverse program to develop a website for the Zambia basketball all-star weekend. As per project requirements, kindly see [the project documentation video](https://www.loom.com/share/541483dbb87f43c1826af4b27364fbd0) demonstrating the added features. Furthermore, below is a tabulated and detailed summary of the features developed.


- Demonstrate ability to create web and mobile User Interfaces adaptable to different screen sizes using CSS media queries.

- Personalized website content inspired by the recommended design by applying instructed design patterns such as colors, layout composition, space between elements, typography font face, size, and weight.

- Used JavaScript Event listeners to toggle menu visibility on mobile devices.

- Applied JavaScript DOM manipulation principle to dynamically render featured speakers' data from a JavaScript Object.

- Applied best semantic principles for improved accessibility.

- Deployed project for public usage using GitHub pages

- Used GitFlow and linters to maximize best code practices and consistency.

mandelatuks committed:
Add index.html,css,main.js files and necessary linters

mandelatuks committed:
Add social link and navigation menu to both home and about pages

mandelatuks committed:
Add main program section with details

mandelatuks committed:
Add speakers section,footer section and started working on about page

mandelatuks committed:
Add past event section

mandelatuks committed:
Add past event place and their date

mandelatuks committed:
Add a footer for about page

mandelatuks committed:
Add styling for mobile version

mandelatuks committed:
Add styling for mobile version on program section

mandelatuks committed:
Add styling for featured speakers section

mandelatuks committed:
Add styling for the footer and started the about page

mandelatuks committed:
Add styling for the logo area

mandelatuks committed:
Add styling for past event and footer to the about page

mandelatuks committed:
Add styling for home in desktop version

mandelatuks committed:
Add styling for home footer and parterns

mandelatuks committed:
Add styling for for home page and about page including sections

mandelatuks committed:
modifie home page and removed the speaker section and rendered it dynamically using js

mandelatuks committed:
Add humberger menu and styled it and also js function for it

mandelatuks committed:
Add some little changes on home page and mobile version

mandelatuks committed:
Add some little changes the footer

mandelatuks committed:
Fix humberger picture

ShoiraTa commented on index.html:
```diff
@@ -0,0 +1,196 @@
+        </div>
+  </header>
+ 
+    <section class="info-head">
+        <p class="hello-text">"Hello! sharing The World"</p>
```
- [ ] The content on the header is hardly readable. I recommend adding a dark or bright filter on top of the image so that you have more contrast between the letters and background color. An example of adding gradient in css with bg image: 

`background-image: linear-gradient( rgba(0,0,0,0.4),    rgba(0,0,0,1)  ), url(../images/basketball2image.jpg)`

Whoistolu replied:
- [ ] [REQUIRED] As required by the previous reviewer, kindly ensure the accessibility look great. in order to enhance a good UI. Currently, the font color you applied to the `h1` tag  `#ec5242` and the green color of the background image conflict with each other.
To fix this: Reduce the opacity of the background image or implement the method the last reviewer recommended above.
### Your Implementation
![Screenshot from 2022-04-14 22-29-55](https://user-images.githubusercontent.com/58692590/163481254-2ebfd8fc-1303-40bd-8f00-4e25a5e31635.png)

ShoiraTa commented on about.html:
```diff
@@ -0,0 +1,169 @@
+</nav>
+
+    <nav class="desktop-nav">
+      <a href="/index.html"> <img src="images/Zambia_BBall-Association.jpg" alt="logo zambia basketball" class="logo-image3"> </a>
```
- [ ] This link is broken because `<a>` tag has wrong path, please delete `slash` (/) before `index.html` and the link will work.

ShoiraTa commented on index.html:
```diff
@@ -0,0 +1,196 @@
+        
+       
+      <nav class="desktop-nav">
+        <a href="/about.html"> <img src="images/Zambia_BBall-Association.jpg" alt="logo zambia basketball" class="logo-image3"></a>
```
- [ ] This link is broken because `<a>` tag has wrong path, please delete `slash` (/) before `index.html` and the link will work.

ShoiraTa commented on index.html:
```diff
@@ -0,0 +1,196 @@
+
+    <section class="main-progrmas">
+      <div class="main-header">
+      <h2 class="program-header1"> Main Program</h2>
+      <hr>
+    </div>
```
- [ ] In order to properly align content inside the cards of `Main Program` section, I suggest using grid:

With Grid  | With flexbox
------------- | -------------
<img width="362" alt="Screenshot 2022-04-14 at 17 49 30" src="https://user-images.githubusercontent.com/77038610/163404681-0873a35a-3453-4b65-95f6-595ef80a0e57.png"> | <img width="366" alt="Screenshot 2022-04-14 at 17 48 38" src="https://user-images.githubusercontent.com/77038610/163404513-3ae4c6a9-5636-41fd-a00c-18d81a6354af.png">

All I have done is: 

```
.cards-program {
   display: grid;
    grid-template-columns: 20% 30% 50%;
    gap: 10px;
}
```

You should only adjust width and margins  so that it is responsive.

Whoistolu replied:
- [ ] [REQUIRED] As recommended by the previous reviewer, kindly ensure the contents of the cards are properly aligned. Presently, the `p` tag contents are going out of the cards and the and `h2` tag contents is not aligned to the center. 

![Screenshot from 2022-04-14 21-49-30](https://user-images.githubusercontent.com/58692590/163473400-973af3df-a509-4296-9b5a-a65c6461f983.png)

mandelatuks replied:
@Whoistolu  kindly check the changes I have made 
Added width and margins according to the specification

![Screenshot 2022-04-14 232622](https://user-images.githubusercontent.com/38649067/163480321-62cc93cd-0164-492f-9a5d-11a4551b4dd4.png)

ShoiraTa commented on index.html:
```diff
@@ -0,0 +1,196 @@
+        </article>
+         
+        <div class="btn-join">
+         <a class="join-button">Join All star weekend here 2022</a>
```
- [ ] The content inside the button is wrapping in mobile size, I suggest leaving only `Join` text inside the button:  

<img width="416" alt="Screenshot 2022-04-14 at 17 58 43" src="https://user-images.githubusercontent.com/77038610/163406476-a26df0b1-df21-44ef-89de-5edbbed96f5e.png">

Whoistolu replied:
Now fixed 👍🏽

ShoiraTa commented on css/styles.css:
```diff
@@ -0,0 +1,855 @@
+  margin-left: 20px;
+}
+
+.hello-text {
+  width: inherit;
```
- [ ] In this case, `width: inherit;` is being a reason of whitspace on the right side of the document on mobile width, I suggest to delete it:

https://user-images.githubusercontent.com/77038610/163408319-a1e9bdac-80d9-4d8f-9645-0930c1fddb4a.mp4

Whoistolu replied:
Now fixed 👍🏽

ShoiraTa commented on about.html:
```diff
@@ -0,0 +1,169 @@
+<body>
+  <header class="head">
+    <nav class="navigation">
+      <a id="bar" class="hum" href="#">
+        <img src="images/menu-trigger-1.png" alt="humberger image">
```
- [ ] The background of the hamburger image should be transparent, please delete the white space on top of the document:
<img width="426" alt="Screenshot 2022-04-14 at 18 11 29" src="https://user-images.githubusercontent.com/77038610/163408715-986ddb64-bddf-4bf2-bdcf-a353495559bf.png">

Whoistolu replied:
- [ ] [REQUIRED] As required by the previous reviewer, you have done a great implementation but the UI still does not enhance good accessibility yet.  Currently, the hamburger is not visible due to the color combination of the background and the `h1` tag.
To fix this: Kindly ensure there is a margin-top on the `h1` tag and reduce the opacity of the background image as it looks too solid/high.

Your Implementation   |  Figma Design
:-------------------------:|:-------------------------:
![Screenshot from 2022-04-14 22-01-50](https://user-images.githubusercontent.com/58692590/163476752-71758dcd-04dc-4ff1-b980-10078890f609.png) | ![Screenshot from 2022-04-14 22-03-20](https://user-images.githubusercontent.com/58692590/163476802-b878b1a2-6a93-4c98-a1b7-4f68d659fdbe.png)

ShoiraTa commented on about.html:
```diff
@@ -0,0 +1,169 @@
+
+  </section>
+     
+  <section class="paterns-about">
+    <h3 class="paterns-info"> Paterns</h3>
+    <hr>
```
- [ ] Please fix the typo `Paterns => Partners` and add some space on the top of the header:
<img width="533" alt="Screenshot 2022-04-14 at 18 14 42" src="https://user-images.githubusercontent.com/77038610/163409267-875367cb-86b8-47ac-9d69-f57285387c42.png">

Whoistolu replied:
- [ ] [REQUIRED] As required by the previous reviewer, you fixed the typo error on the desktop but it remains on the `About` section of the mobile version. Kindly fix it.

### Your Design
![Screenshot from 2022-04-14 22-25-29](https://user-images.githubusercontent.com/58692590/163479910-6f92302e-68df-4fdc-9f97-b430d07dffa8.png)

ShoiraTa commented on about.html:
```diff
@@ -0,0 +1,169 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
```
- [ ] There is some whitespace overflow on the right side, please delete it. You can do that by simply deleting:
```
#downpart {
    width: 100%;
}
```
and
```
.hello-text {
    width: inherit;
} 
```
<img width="294" alt="Screenshot 2022-04-14 at 18 19 53" src="https://user-images.githubusercontent.com/77038610/163410231-ac8166e5-0381-415f-8789-e6831af57112.png">

Whoistolu replied:
This has been fixed 👍🏽

ShoiraTa commented on JS/main.js:
```diff
@@ -0,0 +1,118 @@
+// const humbergerBtn = document.querySelector('.hamburger');
+// const closeButton = document.querySelector('.closeb');
+// const navMenu = document.querySelector('.nav-menu');
+// const navItems = document.querySelectorAll('.nav-link');
+
+// function menuToglling() {
+//   if (navMenu.classList.contains('menuShow')) {
+//     navMenu.classList.remove('menuShow');
+//     closeButton.style.dislay = 'none';
+//     humbergerBtn.style.dislay = 'block';
+//     navMenu.style.dislay = 'none';
+//   } else {
+//     navMenu.classList.add('menuShow');
+//     closeButton.style.dislay = 'block';
+//     humbergerBtn.style.dislay = 'none';
+//     navMenu.style.dislay = 'block';
+//   }
+// }
+
+// humbergerBtn.addEventListener('click', menuToglling);
+// closeButton.addEventListener('click', menuToglling);
+// navItems.forEach((navItem) => {
+//   navItem.addEventListener('click', menuToglling);
+// });
```
- [ ] Please delete all commented lines to keep all files clean.

Whoistolu replied:
This has been fixed 👍🏽

ShoiraTa reviewed (changes requested):
STATUS: CHANGES REQUIRED 🟡 
----
Hi @mandelatuks ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

There are some issues that you still need to work on to go to the next project but you are almost there!

![woohoo-youre-almost](https://user-images.githubusercontent.com/77038610/154921145-1f7bb8f6-c843-4ba5-a534-6b840c600558.jpeg)


 ## Required Changes  :recycle: 

- [ ] The provided live link to project in readme does not work. Please fix the link, you can start from checking which branch is deployed on the gh-pages. If it is main, please manually change it to current branch. 

- [ ] Please make a habit to use `gitflow` while developing. You should change the `main` branch to `development` branch. So that your Pull request is from `project-capstone` branch to `development`

_Check the comments under the review._ 

## Optional suggestions

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mandelatuks committed:
modifie files according to the review requirement

mandelatuks committed:
Update README.md

Add a working link for website

mandelatuks committed:
Merge branch 'Development' into project-capstone

mandelatuks commented:
@ShoiraTa 
# kindly check the pictures added down here according to the change that was required 

- the following were done 
- the overflow was removed 
- change the typo for patterns to collect one and add the collecting padding
- The main program cards were displayed using a grid 
- removed the background white on about page on top
- removed the necessary code in the js file for clean code purpose 
- update the right link for the deployed website
- removed the dead on desktop

 
![Screenshot 2022-04-14 205622](https://user-images.githubusercontent.com/38649067/163461954-14981dea-9b42-4610-a00e-a561392ca78e.png)
![Screenshot 2022-04-14 205856](https://user-images.githubusercontent.com/38649067/163461977-2e5eef43-29d7-41e8-9fd2-4b451d475b73.png)
![Screenshot 2022-04-14 210023](https://user-images.githubusercontent.com/38649067/163461980-bfc2919a-62c0-4726-a694-1c6f1efbcde3.png)
![Screenshot 2022-04-14 210048](https://user-images.githubusercontent.com/38649067/163461983-c9e46390-0ff1-4dac-aaea-dab98d9103b9.png)
![Screenshot 2022-04-14 210409](https://user-images.githubusercontent.com/38649067/163461985-4524d64e-78bf-4fef-8ecc-12909e02a48f.png)
![Screenshot 2022-04-14 215640](https://user-images.githubusercontent.com/38649067/163466169-52603381-153c-4fce-b11d-fd616b640fee.png)

ShoiraTa commented:
Hi @mandelatuks , it looks like you did a really great job within such a short time 👏🏻

mandelatuks commented:
@ShoiraTa, it was all due to the proper guidance that you provided towards what I should improve on and what I appreciate.

Whoistolu commented on JS/main.js:
```diff
@@ -0,0 +1,91 @@
+
+speakersEvent.forEach((guest) => {
+  if (featuredSpeakers) {
+    featuredSpeakers.innerHTML += `<article class="event-speakers">
```
- [ ] [OPTIONAL] Great implementation concerning the event speakers, however, for a better UI, we can resize the images as they currently look distorted.


Your Implementation   |  Figma Design
:-------------------------:|:-------------------------:
![Screenshot from 2022-04-14 21-54-29](https://user-images.githubusercontent.com/58692590/163473990-0ff0a9ef-f5dc-43c4-a286-9e4e08afff33.png) | ![Screenshot from 2022-04-14 21-58-12](https://user-images.githubusercontent.com/58692590/163474103-f4898788-3d40-4476-9fcd-1b677dc74385.png)

mandelatuks replied:
@Whoistolu 
for this one its due to the responsive the more you reduce the screen the more they adopt

mandelatuks replied:
![Screenshot 2022-04-14 235910](https://user-images.githubusercontent.com/38649067/163483453-271911a5-17c7-445a-ac96-72acb69d2707.png)

Whoistolu commented on about.html:
```diff
@@ -0,0 +1,169 @@
+    <p class="past-event">
+      Take a look at the last two ZBA all weekend wich took place in Livingstone and Ndola city.
+    </p>
+    <div class="event-wrapper">
+      <article class="event1">
+          <h2>2020</h2>
+          <p>All start weekend Livingstone, Zambia</p>
+      </article>
+      <article class="events2">
+          <h2>2021</h2>
+          <p>All star weekend ndola city,Zambia</p>
+      </article>
+  </div>
```
- [ ] [OPTIONAL] Kindly ensure that your design looks more like the original design provided. Please ensure you add images to the two cards and an overlay.

Your Implementation   |  Figma Design
:-------------------------:|:-------------------------:
![Screenshot from 2022-04-14 22-15-28](https://user-images.githubusercontent.com/58692590/163478699-45e31620-9b58-48f4-b95a-4ca480db1042.png) | ![Screenshot from 2022-04-14 22-15-43](https://user-images.githubusercontent.com/58692590/163478767-1d93d1a1-5478-47c4-bb84-70fbcde2913e.png)

mandelatuks committed:
change from the reviewer

mandelatuks committed:
Merge branch 'project-capstone' of github.com:mandelatuks/Microverse-First-Capstone into project-capstone

mandelatuks commented:
Hi, Ajise Toluwase
kindly check the update thank you

On Thu, Apr 14, 2022 at 10:54 PM Ajise Toluwase ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In index.html
> <https://github.com/mandelatuks/Microverse-First-Capstone/pull/1#discussion_r850796908>
> :
>
> > +      <h2 class="program-header1"> Main Program</h2>
> +      <hr>
> +    </div>
>
>
>    - [REQUIRED] As recommended by the previous reviewer, kindly ensure
>    the contents of the cards are properly aligned. Presently, the p tag
>    contents are going out of the cards and the and h2 tag contents is not
>    aligned to the center.
>
> [image: Screenshot from 2022-04-14 21-49-30]
> <https://user-images.githubusercontent.com/58692590/163473400-973af3df-a509-4296-9b5a-a65c6461f983.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mandelatuks/Microverse-First-Capstone/pull/1#discussion_r850796908>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJG3Z25UAUFT7JE6KF45OPTVFCAXDANCNFSM5TNXIC2Q>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mandelatuks committed:
change from the reviewer

Whoistolu reviewed (changes requested):
Hi @mandelatuks 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mandelatuks committed:
change from the reviewer

mandelatuks commented:
@Whoistolu kindly check the changes made

![Screenshot 2022-04-15 003516](https://user-images.githubusercontent.com/38649067/163489953-909fe8d3-d38e-4739-96df-f189d630ae21.png)
![Screenshot 2022-04-15 003614](https://user-images.githubusercontent.com/38649067/163489960-17f190b9-304b-402a-a084-de4871736a95.png)
![Screenshot 2022-04-15 003700](https://user-images.githubusercontent.com/38649067/163489962-1f1f698f-d46e-4e40-b8ce-5e4e1f542ac0.png)
![Screenshot 2022-04-15 003742](https://user-images.githubusercontent.com/38649067/163489965-ed04dd76-db7c-4d9a-b7e1-8905a926f40b.png)
